src/Components/Menu/MenuItem.js

```diff
@@ -6,16 +6,20 @@ import './MenuItem.css';
 export default function MenuItem(props) {
   const ctx = useContext(ItemsContext)
 
-  const [newCartItem,setNewCartItem] =useState(props.menuItem)
+  const [newCartItem,setNewCartItem] =useState(props.menuItem);
 
   function qtyHandler (e) {
-    setNewCartItem((prev) =>
-      ({...prev, itemQty: +e.target.value})
-    )
+    if (e.target.value >= 0) {
+      setNewCartItem((prev) => ({
+        ...prev, itemQty: +e.target.value
+      }))
+    } else {
+      setNewCartItem(props.menuItem);
+    }
```

Comment:
after incrementing if we again decrement to zero,The product in cart item wont get removed because it is already added.I am updating a remove function in context file so, that you can remove the product on any condition by a function call.